documentation/internal/Development-Workflow.md

```diff
@@ -0,0 +1,311 @@
+
+**Rule 3**: Use the **Pattern**:
+
+- Query: `get` `Entity` `.` `query` `.ts`
+- Mutation: (`create` | `update`) `Entity` `.` `mutation` `.ts`
+- Fragment: (`add` | `remove`) `Entity` `.` `fragment` `.ts`
+
+##### fragments
+
+_What is a fragment?_ A partial query or mutation that can be used as part of other queries or (Mainly) for getting or setting some data to the cache. We do have to update the cache after a mutation to keep it up-to-date.
+
+##### queries
+
+_What is a query?_ Something to be fetched from the GraphQL server. Can have some input and defines what is returned.
+
+##### mutation
+
+_What is a mutation?_ Something to be posted to the GraphQL server. Habe some defined input and also something that is returned.
+
+#### helpers
+
+Add here any methods that can be used in other places - that aren't hooks or compnents itselves.
+
+#### hooks
+
+Our **custom hooks** - exaplained in more detail in the Development sections below.
+
+### Files on the root src folder
+
+- `cache.tsx` - Start the Apollo Cache.
+  Also define some local state to be used within the cache on the root level or as part of any table (@client). Read more [here](https://www.apollographql.com/docs/react/local-state/managing-state-with-field-policies/#storing-local-state-in-the-cache).
+
+- `config.json` - Some global varibles to be used during the configuration of the App. i.e: GraphQL server host.
+
+- `index.tsx` - App start - Just connects the Apollo Cache (from `cache.tsx`), add semantic-ui for UI and call the main component: `App.tsx`.
+
+---
+
+# Local vs. Remote states
+
+### Remote states
+
+The local data kept in the app by Apollo client CACHE is considered remote state since is a small mirror of the GraphQL server. The cached data is reused for up-coming queries (if we don't enforce a `refetch` from the server), just returning the local **remote state** for the queried data - when it is available. Or get from the server if not available in cache.
+
+**Important**: No need to do anything different to query from the cache instead of the server. Just a simple query will do it. Mutations on the other hand require some addition - after sending data to the server we have to update the cache.
+
+_How Apollo keeps the cache synced with the server?_
+
+Apollo client provides this cache management module, which will store all the data query or mutation results after it is fetched from the server. It requires you add the `id` of each entity - and the related entities included in the fetched data - or another identifier that you have specified before to the cache management instance if not using the `id`. The caches stores one instance of each entity queried (refreshing would clear the cache). And for each relationship it keeps a reference to its entity. In the end the remote states has one instance of each entity already fetched from the server.
```

Comment:
So it doesn't need unique global id? Table and id is sufficient for it to be identified?

Reply:
Yes. The id should be unique between entities in the same table, that's why by default it uses the table `id` field. :)